### DISCUSSION TITLE
How to custom rowKey for composite index in Hbase

### DISCUSSION THREAD

**Discussion by dinhphu2k1-gif** [JanusGraph/janusgraph#3329] [Q&A]
I want to custom rowKey for composite index when it's inserted to Hbase table by add salt to rowKey. 
I find a post in StackOverFlow about this problem https://stackoverflow.com/questions/45237151/hbase-row-key-split-algorithm.
But they don't show where  i have to modify code or config Hbase.

Can anyon help me?

**Comment by vtslab** [comment]
What do you mean with the "rowKey for composite index" and what improvement should a custom rowKey bring?

**Reply by dinhphu2k1-gif** [reply]
I don't know if the composite index is evenly distributed across regions, so I'd like to prefix the key (composite index includes key-value) according to the stackoverflow post to make sure it's evenly distributed. Because I see the amount of requests across regions is not uniform

**Comment by vtslab** [comment]
OK, it sounds like you mean [graph partitioning](https://docs.janusgraph.org/advanced-topics/partitioning/#random-vs-explicit-partitioning), that is the placement of vertices with their adjacencies across partitions/regions. Although the composite index is also stored in HBase as a separate table, it seems unlikely that you see the effect of unevenly spread index lookups.

For implementing a custom placement strategy the ids.num-partitions and ids.placement config properties are relevant, as well as the [IDPlacementStrategy interface](https://javadoc.io/static/org.janusgraph/janusgraph-core/0.6.2/org/janusgraph/graphdb/database/idassigner/placement/IDPlacementStrategy.html) to implement.

**Reply by dinhphu2k1-gif** [reply]
@vtslab  so How do i fix it (amount of requests across regions is not balance)?
